front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -115,6 +115,23 @@ class PipelineController {
     pipelineDAO.update(pipelineId, pipeline)
   }
 
+  @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
+  Pipeline update(@PathVariable String id, @RequestBody Pipeline pipeline) {
+    Pipeline existingPipeline = pipelineDAO.findById(id)
+    if (!existingPipeline || pipeline.id != id) {
+      throw new InvalidPipelineRequestException()
+    }
+
+    if (pipelineDAO.getPipelinesByApplication(pipeline.getApplication()).any {
+      it.getName().equalsIgnoreCase(pipeline.getName()) && it.getId() != id }) {
+      throw new DuplicatePipelineNameException()
+    }
+
+    pipeline.updateTs = System.currentTimeMillis()
+    pipelineDAO.update(id, pipeline)
+    return pipeline
+  }
+
   static class RenameCommand {
```

Comment:
And `RenameCommand` will disappear at some point when the callers are switched over to the `PUT`?